src/Libraries/CoreNodes/Data.cs

```diff
@@ -528,6 +529,148 @@ public static bool CanObjectBeCached(object inputObject)
         }
 
         internal static DynamoLogger dynamoLogger = ExecutionEvents.ActiveSession?.GetParameterValue(ParameterKeys.Logger) as DynamoLogger;
+        #endregion
+
+        #region Input Output Node
+
+        public class DataNodeDynamoType
+        {
+            public Type Type { get; private set; }
+            public string Name { get; private set; }
+            public DataNodeDynamoType Parent { get; private set; }
+
+            public DataNodeDynamoType(Type type, string name = null)
+            {
+                Type = type;
+                Name = name ?? type.Name;
+            }
+
+            public DataNodeDynamoType(Type type, DataNodeDynamoType parent, string name = null)
+            : this(type, name)
+            {
+                Parent = parent;
+            }
+        }
+
+
+        /// <summary>
+        /// A static dictionary for all Dynamo supported data types
+        /// </summary>
+        /// <returns>The dictionary containing the supported data types</returns>
+        public static List<DataNodeDynamoType> GetDataNodeDynamoTypeList()
```

Comment:
why is this public?

Reply:
this method seems inefficient, whenever this method is called it re evaluates allocating this list and allocating all these types... seems this could be done at compile time using an initializer - though I am not sure.